### PR TITLE
8306067: Open source AWT Graphics,GridBagLayout related tests

### PR DESCRIPTION
**Pull request by prsadhuk**
This opensources few AWT Graphics & GridBagLayout tests

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8306067](https://bugs.openjdk.org/browse/JDK-8306067): Open source AWT Graphics,GridBagLayout related tests

### Reviewers
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13570/head:pull/13570` \
`$ git checkout pull/13570`

Update a local copy of the PR: \
`$ git checkout pull/13570` \
`$ git pull https://git.openjdk.org/jdk.git pull/13570/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13570`

View PR using the GUI difftool: \
`$ git pr show -t 13570`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13570.diff">https://git.openjdk.org/jdk/pull/13570.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13570#issuecomment-1517230018)

### CODE REVIEW

File: test/jdk/java/awt/Graphics/GetGraphicsTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * @summary Checks getGraphics doesn't throw NullPointerExcepton for invalid colors and font.
+ * @run main GetGraphicsTest
+*/
+import java.awt.image.BufferedImage;
+import java.awt.Color;
+import java.awt.EventQueue;
+import java.awt.Font;
+import java.awt.Graphics;
+
+public class GetGraphicsTest {
```
Comment (mrserb) [inline test/jdk/java/awt/Graphics/GetGraphicsTest.java:35]: What this test is expected to test? the getBackground/getForeground/getFont seems to be all unused.

File: test/jdk/java/awt/GridBagLayout/GridBagLayoutButtonsOverlapTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        c.fill = GridBagConstraints.BOTH;
+        c.weightx = 1.0;
+        b1 = makeButton("button1", gridbag, c);
+        b2 = makeButton("button2", gridbag, c);
+
+        c.gridwidth = GridBagConstraints.REMAINDER;
+        b4 = makeButton("button4", gridbag, c);
+
+        c.weightx = 0.0;
+        b5 =  makeButton("button5", gridbag, c);
```
Comment (turbanoff) [inline test/jdk/java/awt/GridBagLayout/GridBagLayoutButtonsOverlapTest.java:81]:
```suggestion
        b5 = makeButton("button5", gridbag, c);
```

File: test/jdk/java/awt/GridBagLayout/GridBagLayoutButtonsOverlapTest.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        b10 = makeButton("button10", gridbag, c);
+    }
+
+    public static void test() {
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+        frame.validate();
+
+        int b1Corner = b1.getLocation().y + b1.getHeight();
+        int  b5Corner = b5.getLocation().y;
```
Comment (turbanoff) [inline test/jdk/java/awt/GridBagLayout/GridBagLayoutButtonsOverlapTest.java:107]:
```suggestion
        int b5Corner = b5.getLocation().y;
```

File: test/jdk/java/awt/GridBagLayout/GridBagLayoutButtonsOverlapTest.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        if (b1Corner > b5Corner) { //they are equals each other in best case
+            throw new RuntimeException("Buttons are overlapped when container is small enough");
+        }
+        System.out.println("Test passed.");
+    }
+
+    protected static Button  makeButton(String name,
```
Comment (turbanoff) [inline test/jdk/java/awt/GridBagLayout/GridBagLayoutButtonsOverlapTest.java:114]:
```suggestion
    protected static Button makeButton(String name,
```

File: test/jdk/java/awt/GridBagLayout/GridBagLayoutButtonsOverlapTest.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                                 GridBagLayout gridbag,
+                                 GridBagConstraints c) {
+        Button button  = new Button(name);
```
Comment (turbanoff) [inline test/jdk/java/awt/GridBagLayout/GridBagLayoutButtonsOverlapTest.java:117]:
```suggestion
        Button button = new Button(name);
```

File: test/jdk/java/awt/GridBagLayout/GridBagLayoutOutOfBoundsTest.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    public static void main(String[] args) throws Exception {
+        EventQueue.invokeAndWait(() -> {
+            try {
+                frame = new Frame("GridBagLayoutOutOfBoundsTestFrame");
+                frame.validate();
+                GridBagLayout layout = new GridBagLayout();
+                frame.setLayout(layout);
+                GridBagConstraints gridBagConstraints;
+
+                Button[]  mb = new Button[L];
```
Comment (turbanoff) [inline test/jdk/java/awt/GridBagLayout/GridBagLayoutOutOfBoundsTest.java:51]:
```suggestion
                Button[] mb = new Button[L];
```

File: test/jdk/java/awt/Graphics/GetGraphicsTest.java (hunk 1 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Graphics/GetGraphicsTest.java:35]: As the test summary mentioned, it supposed to "Check getGraphics doesn't throw NullPointerExcepton for invalid colors and font"...the closed test was also like it...was applet based which is just converted to use BufferedImage.getGraphics to be a headless test..
Seems [4746122](https://bugs.openjdk.org/browse/JDK-4746122):Failed to load applet fix is about applet,
Once comment in the bug says 
```
The problem is applet overrides Component.getBackground()/getForeground() to
return null. So SunGraphics2D is created with null params
```

 so I guess this test is redundant now and we can remove..please suggest..

Comment (mrserb) [reply test/jdk/java/awt/Graphics/GetGraphicsTest.java:35]: Probably we may try to extend the Frame by the GetGraphicsTest, and then get a graphics from it?

Comment (prsadhuk) [reply test/jdk/java/awt/Graphics/GetGraphicsTest.java]: ok

**Review by mrserb — APPROVED** [review]

**Pull request closed without merging**